### DISCUSSION TITLE
plan for version 3 major release

### DISCUSSION THREAD

**Discussion by dongvng** [TanStack/virtual#494] [Q&A]
Hi maintainers,

Do you have a timeline to release major version 3 since the beta has been developing for quite a year?

**Comment by piecyk** [comment]
Hi @dongvng 

I think we are pretty close, api looks stable, what we are still missing 

* improve unit tests, add playwright tests because of jsdom's limitations
* maybe add some performance testing
* update dosc/examples

**Comment by mkosir** [comment]
Hi, thanks for putting out v3 beta 🚀 
Any idea how close we are to release/out of beta?

**Comment by nicolaschambrier** [comment]
Hello 👋 
We're trying to keep our dependencies up to date but it's not comfortable adding a beta version... What should we do? We're now more than a year late 😭

**Comment by abdullaev388** [comment]
What happened to this lib? Last commit was in March, is it abandoned?

**Comment by tbertrand7** [comment]
Any update on this? This is the one dependency I have blocking me from upgrading to React 18.

**Comment by itsjavi** [comment]
The last beta seems pretty stable to me. Any plans to promote it to v3 final?

Also the libraries in the packages.json seem pretty outdated (like Typescript v4)

@tannerlinsley @piecyk

**Comment by GitMurf** [comment]
Would love to know as well. Thanks for the great work Tanner and team!

**Comment by abradeatlas** [comment]
Any updates on stable release version?

**Comment by GitMurf** [comment]
See here for update from a couple weeks ago: https://github.com/TanStack/virtual/discussions/570#discussioncomment-6858713
